doc/source/whatsnew/v0.23.2.txt

```diff
@@ -61,6 +61,7 @@ Bug Fixes
 
 - Bug in :meth:`Index.get_indexer_non_unique` with categorical key (:issue:`21448`)
 - Bug in comparison operations for :class:`MultiIndex` where error was raised on equality / inequality comparison involving a MultiIndex with ``nlevels == 1`` (:issue:`21149`)
+- The backtrace from a ``KeyError`` is now shorted and clearer (:issue:`21557`)
```

Comment:
can you be a little more clear here :> (this is when indexing!)

backtrace -> traceback

Reply:
I think this should go in 0.24 as this might require a tricky backport (maybe) as the refactored code is in 0.24